src/dense/generic_dense.jl

```diff
@@ -227,18 +219,18 @@ function ode_interpolation(tvals,id,idxs,deriv,p,continuity::Symbol=:left)
     Θ = iszero(dt) ? oneunit(t) / oneunit(dt) : (t-ts[i₋]) / dt
 
     if typeof(cache) <: (FunctionMapCache) || typeof(cache) <: FunctionMapConstantCache
-      vals[j] = ode_interpolant(Θ,dt,timeseries[i₋],timeseries[i₊],0,cache,idxs,deriv)
+      return ode_interpolant(Θ,dt,timeseries[i₋],timeseries[i₊],0,cache,idxs,deriv)
     elseif !id.dense
-      vals[j] = linear_interpolant(Θ,dt,timeseries[i₋],timeseries[i₊],idxs,deriv)
+      return linear_interpolant(Θ,dt,timeseries[i₋],timeseries[i₊],idxs,deriv)
     elseif typeof(cache) <: CompositeCache
       DiffEqBase.addsteps!(ks[i₊],ts[i₋],timeseries[i₋],timeseries[i₊],dt,f,p,cache.caches[id.alg_choice[i₊]]) # update the kcurrent
-      vals[j] = ode_interpolant(Θ,dt,timeseries[i₋],timeseries[i₊],ks[i₊],cache.caches[id.alg_choice[i₊]],idxs,deriv)
+      return ode_interpolant(Θ,dt,timeseries[i₋],timeseries[i₊],ks[i₊],cache.caches[id.alg_choice[i₊]],idxs,deriv)
     else
       DiffEqBase.addsteps!(ks[i₊],ts[i₋],timeseries[i₋],timeseries[i₊],dt,f,p,cache) # update the kcurrent
-      vals[j] = ode_interpolant(Θ,dt,timeseries[i₋],timeseries[i₊],ks[i₊],cache,idxs,deriv)
+      return ode_interpolant(Θ,dt,timeseries[i₋],timeseries[i₊],ks[i₊],cache,idxs,deriv)
     end
   end
-
+  vals = _vals[sortperm(idx)]
```

Comment:
Is `sortperm` really needed here? Isn't this already handled by https://github.com/mohamed82008/OrdinaryDiffEq.jl/blob/398d5263c93098538ec6e20c799357af4fd47587/src/dense/generic_dense.jl#L198 (which also takes care of the integration direction)?

Reply:
The sortperm here is on `idx` to get the "inverse" of `idx` that will put the elements of `_vals` in the original order they should be in. Previously we would just allocated at the proper index `j`. Now I use a map so the order is not right.

Reply:
Ah, I see. This additional allocation and sorting is unfortunate.

Reply:
I guess it might also be more efficient to use
```suggestion
  vals = _vals[invperm(idx)]
```
here.